pymess/backend/emails/__init__.py

```diff
@@ -147,10 +151,15 @@ def send_batch(self):
         sent = 0
 
         try:
-            waiting_emails_qs = EmailMessage.objects.filter(
-                state=EmailMessage.STATE.WAITING
+            messages_to_send = EmailMessage.objects.filter(
+                Q(state=EmailMessage.STATE.WAITING) |
+                Q(
+                    state=EmailMessage.STATE.ERROR,
+                    send_attempts_count__lte=settings.EMAIL_BATCH_SEND_ATTEMPTS_COUNT,
+                    created_at__gte=now() - timedelta(seconds=settings.EMAIL_BATCH_MAX_SECONDS_TO_SEND)
```

Comment:
If I understand this correctly, it will not work as intended during time change (ie to/from daylight saving time) - but that is probably an edge case.